index.d.ts

```diff
@@ -7,7 +7,7 @@ export interface Options {
     timeCost?: number;
     memoryCost?: number;
     parallelism?: number;
-    argon2d?: boolean;
+    type?: 0 | 1 | 2;
```

Comment:
AFAIK there is no syntax that could take a constant like this. How about this approach?

```typescript

type argon2dType = 0;
type argon2iType = 1;
type argon2idType = 2;

export interface Options {
    hashLength?: number;
    timeCost?: number;
    memoryCost?: number;
    parallelism?: number;
    type?: argon2dType | argon2iType | argon2idType;
}

...

export const argon2d: argon2dType;
export const argon2i: argon2iType;
export const argon2id: argon2idType;

...
```
But this does not change anything, just adds an alias for 0, 1 and 2. Maybe the right way would be to use a typescript Enum along with a JS object in the index.js? But this would change the type mode to:
```javascript
const defaults = Object.freeze({
  hashLength: 32,
  timeCost: 3,
  memoryCost: 12,
  parallelism: 1,
  type:  argon2Type.argon2i
})
```